pkg/apis/syn/v1alpha1/cluster_types.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// FactKey is the key for the facts map
+type FactKey string
+
+// FactValue is the value for the facts map
+type FactValue string
+
+// Facts is a map of arbitrary facts for the cluster
+type Facts map[FactKey]FactValue
+
+// ClusterSpec defines the desired state of Cluster
+type ClusterSpec struct {
+	// APIEndpointSecretRef references the secret containing connection information to Kubernetes API endpoint of the registered Kubernetes cluster.
+	APIEndpointSecretRef *corev1.SecretKeySelector `json:"apiEndpointSecretRef,omitempty"`
```

Comment:
A `SecretKeySelector` only selects a single key of a secret, instead of the whole secret.
```suggestion
	APIEndpointSecretRef *corev1.SecretReference `json:"apiEndpointSecretRef,omitempty"`
```

Reply:
Assumes from SDD -> should be specified there as well